hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1378,47 +1434,71 @@ func (r *NodePoolReconciler) getTunedConfig(ctx context.Context,
 
 	for _, config := range configs {
 		manifestRaw := config.Data[tunedConfigKey]
-		manifest, err := validateTunedConfigManifest([]byte(manifestRaw))
+		manifest, manifestType, err := validateTunedConfigManifest([]byte(manifestRaw))
 		if err != nil {
-			errors = append(errors, fmt.Errorf("configmap %q failed validation: %w", config.Name, err))
+			errors = append(errors, fmt.Errorf("configmap %q (type:%v) failed validation: %w", config.Name, manifestType, err))
 			continue
 		}
+		if manifestType == Tuned {
+			tunedAllConfigPlainText = append(tunedAllConfigPlainText, string(manifest))
+		} else if manifestType == PerformanceProfile {
+			performanceProfileAllConfigPlainText = append(performanceProfileAllConfigPlainText, string(manifest))
+		}
+	}
 
-		allConfigPlainText = append(allConfigPlainText, string(manifest))
+	if len(performanceProfileAllConfigPlainText) > 1 {
+		errors = append(errors, fmt.Errorf("there can NOT be more than ONE PerformanceProfile per NodePool. found: %d", len(performanceProfileAllConfigPlainText)))
 	}
 
 	// Keep output deterministic to avoid unnecesary no-op changes to Tuned ConfigMap
-	sort.Strings(allConfigPlainText)
-	return strings.Join(allConfigPlainText, "\n---\n"), utilerrors.NewAggregate(errors)
+	sort.Strings(tunedAllConfigPlainText)
+	sort.Strings(performanceProfileAllConfigPlainText)
+
+	return tunedConfigData{
+		TunedConfigRaw:              strings.Join(tunedAllConfigPlainText, "\n---\n"),
+		PerformanceProfileConfigRaw: strings.Join(performanceProfileAllConfigPlainText, "\n---\n"),
+		Errors:                      utilerrors.NewAggregate(errors),
+	}
 }
 
-func validateTunedConfigManifest(manifest []byte) ([]byte, error) {
+func validateTunedConfigManifest(manifest []byte) ([]byte, tunedManifestType, error) {
```

Comment:
```suggestion
func validateTunedConfigManifest(manifest []byte) (manifestTuned []byte, manifestPerformanceProfile []byte, error) {
```
Related to my previous comment, I think we could get rid of tunedManifestType by returning `manifestTuned []byte, manifestPerformanceProfile []byte, error`